lib/JSON/Validator.pm

```diff
@@ -586,13 +586,15 @@ sub _validate_all_of {
 
   return if not @errors;
 
+  my @details = map { map $_->details, @{$_}[1 .. $#{$_}] } @errors;
```

Comment:
This is too hard to read. I don't like nested maps, and there's just too much {@[$#]}. (Same below)

Reply:
I'm not sure what I can do with that, as the code is correct now :)  I can add more local variables, which would make the expression longer.. I'm not sure if it would be any easier to read.

Reply:
I'll push a change with some local vars showing how the list is being chopped up and see if that makes it better.